src/app/dataset-view/additional-components/dataset-settings-component/services/dataset-scheduling.service.ts

```diff
@@ -0,0 +1,59 @@
+import { Injectable } from "@angular/core";
+import { ToastrService } from "ngx-toastr";
+import { Observable, map } from "rxjs";
+import { DatasetFlowApi } from "src/app/api/dataset-flow.api";
+import {
+    DatasetFlowBatchingMutation,
+    DatasetFlowScheduleMutation,
+    DatasetFlowType,
+    GetDatasetFlowConfigsQuery,
+    ScheduleInput,
+    TimeDeltaInput,
+} from "src/app/api/kamu.graphql.interface";
+
+@Injectable({
+    providedIn: "root",
+})
+export class DatasetSchedulingService {
+    constructor(private datasetFlowApi: DatasetFlowApi, private toastrService: ToastrService) {}
+
+    public fetchDatasetFlowConfigs(
+        datasetId: string,
+        datasetFlowType: DatasetFlowType,
+    ): Observable<GetDatasetFlowConfigsQuery> {
+        return this.datasetFlowApi.getDatasetFlowConfigs({ datasetId, datasetFlowType });
+    }
+
+    public setDatasetFlowSchedule(params: {
+        datasetId: string;
+        datasetFlowType: DatasetFlowType;
+        paused: boolean;
+        schedule: ScheduleInput;
+    }): Observable<void> {
+        return this.datasetFlowApi.setDatasetFlowSchedule(params).pipe(
+            map((data: DatasetFlowScheduleMutation) => {
+                const setConfigSchedule = data.datasets.byId?.flows.configs.setConfigSchedule;
+                setConfigSchedule?.__typename === "SetFlowConfigSuccess"
+                    ? this.toastrService.success(setConfigSchedule.message)
+                    : this.toastrService.error(setConfigSchedule?.message);
+            }),
+        );
+    }
+
+    public setDatasetFlowBatching(params: {
+        datasetId: string;
+        datasetFlowType: DatasetFlowType;
+        paused: boolean;
+        throttlingPeriod: TimeDeltaInput;
+        minimalDataBatch: number;
```

Comment:
Any of these 2 might be undefined in the API.

Reply:
Fixed